### PR TITLE
implement ice area trap

### PR DESCRIPTION
**Pull request by arthu-rguo**
While a `PlayerMovement` is inside the area, its `friction` is scaled by an adjustable `frictionScale` parameter.

Known issue: The effect is applied when the player's collider enters the trap's bounding box. 
Suggested fix: Add a small sub-collider at the feet of the Player prefab.

### CODE REVIEW

**Review by anjx123 — APPROVED** [review]
Icy fields feels good

**Comment by leo-cf-tian** [comment]
No style issues either, epico

File: Assets/Scripts/TrapScripts/IceZoneTrap.cs
Comment (Hanson-Sun) [inline Assets/Scripts/TrapScripts/IceZoneTrap.cs:1]: what does this file do?

File: Assets/Scripts/PlayerScripts/PlayerMovement.cs
Comment (Hanson-Sun) [inline Assets/Scripts/PlayerScripts/PlayerMovement.cs:1]: resolve current conflicts with master, theres new two player movement stuff

File: Assets/Scripts/TrapScripts/IceZoneTrap.cs
Comment (arthu-rguo) [reply Assets/Scripts/TrapScripts/IceZoneTrap.cs:1]: i forgot to delete it :3

**Pull request merged**